subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/util/ProjectUtil.java

```diff
@@ -0,0 +1,21 @@
+package org.shipkit.internal.gradle.util;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+
+import java.util.Optional;
+
+public class ProjectUtil {
+
+    public static void requireRootProject(final Project project, final Class<?> clazz) {
+        requireRootProject(project, clazz, Optional.empty());
+    }
+
+    public static void requireRootProject(final Project project, final Class<?> clazz, final Optional<String> explanation) {
```

Comment:
Minor: Optional is an overkill in this instance and allowing null is simpler and cleaner.

Optionals are very useful for output parameters (e.g. return types) because they force the clients to code against NPE. For input parameters, they are contrived.